### PR TITLE
emulator: fake warmth | pocketbook: warmth lights support

### LINKED ISSUE #5116: FR: PocketBook frontlight warmth + gestures
Device: Pocketbook Touch HD 3
Version: v2019.06-23

The Touch HD 3 has a frontlight with adjustable light temperature. Sliding in left side up/down to increase/decrease light and sliding in right side up/down to change light temperature (for devices that support this feature) both work on the original reader software and on CoolReader (https://github.com/blchinezu/pocketbook-coolreader), but not in KOReader.

When I try sliding with two fingers it works very slowly and is often interpreted as increasing font size.

### PR DESCRIPTION
**Pull request by pazos**
We all know that the code in `frontend/ui/widget/frontlightwidget.lua` is a hot potato :cry:

I was trying to add support for pocketbook warmth lights but I don't want to copy/paste the mess of auto-warmth again. So this is my humble attempt to have support for warmth gestures without the "auto" thingy.

It works fine on the emulator. It does nothing, just logs the attempt to set intensity and warmth. The same procedure can be used on pocketbook replacing the logs by inkview calls.

I know this is yet another workaround but it is better than nothing.

**Edit:** also added support for Pocketbook warmth lights.

Fixes #5116 

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/6531)

### CODE REVIEW

File: frontend/device/generic/device.lua [hunk 1]
```diff
@@ -41,6 +41,7 @@ local Device = {
     hasFrontlight = no,
     hasNaturalLight = no, -- FL warmth implementation specific to NTX boards (Kobo, Cervantes)
     hasNaturalLightMixer = no, -- Same, but only found on newer boards
+    hasNaturalLIghtApi = no,
```
Comment (pazos) [inline frontend/device/generic/device.lua:44]: typo

**Review by poire-z — COMMENTED** [review]
Looks harmless.
(But my potatology lessons taught me to not trust how a potato look :)

**Comment by pazos** [comment]
Added pocketbook warmth support to this PR to see the whole picture. Only enabled on the device I have here to test. Works fine :)

Pinging @roshavagarga, who probably knows which devices have warmth lights.

**Comment by roshavagarga** [comment]
@pazos Based on a quick check in the PocketBook website:
- Touch Lux 5 (628)
- Touch HD 2 (631) - this one shares the same codename as the Touch HD, which only has the normal Frontlight.
- Touch HD 3 (632)
- Inkpad 3 (740)
- Inkpad 3 Pro (740_2)
- Inkpad X (1040)

**Comment by pazos** [comment]
@roshavagarga: thank you.

I added all of them except the Touch HD 2

**Review by Frenzie — APPROVED** [review]
Looks pretty innocent.

**Comment by roshavagarga** [comment]
@pazos Based off what I could understand for the Touch HD and Touch HD 2 (from manuals) - these devices are entirely the same hardware-wise for the most part, minus the warm light on the HD 2, and they use the exact same firmware files for updates. I'm guessing the software has a hardware check and enables/disables the submenu or whatever's used to control it. If you can find anyone with either of these devices that's opened an issue before, you can ask them for more info on anything that might be used to separate one from the other. Another option is taking a look at the update files for the original firmware and checking for further hints there.

Either way, great job!

**Review by NiLuJe — APPROVED** [review]
What everyone else said ;).

**Comment by pazos** [comment]
> @pazos Based off what I could understand for the Touch HD and Touch HD 2 (from manuals) - these devices are entirely the same hardware-wise for the most part, minus the warm light on the HD 2, and they use the exact same firmware files for updates. I'm guessing the software has a hardware check and enables/disables the submenu or whatever's used to control it. If you can find anyone with either of these devices that's opened an issue before, you can ask them for more info on anything that might be used to separate one from the other. Another option is taking a look at the update files for the original firmware and checking for further hints there.
> 
> Either way, great job!

Thank you again. Based on your suggestion I used the result of `inkview.GetFrontlightColor` to differenciate between models. It should return a value between 0 and 100 if the device is a Touch HD2 and something less than 0 if the device doesn't have warmth lights!

**Comment by roshavagarga** [comment]
@pazos Did you miss the Touch HD 3 or is it just me missing it?

**Comment by pazos** [comment]
> Did you miss the Touch HD 3 or is it just me missing it?

Nope. That's the first device I added because it is the one I have here :)

**Pull request merged**